File: consumer/src/main/java/no/nav/sbl/dialogarena/soknadinnsending/consumer/kontaktinfo/dkif/dto/DigitalKontaktinfoBolk.java
package no.nav.sbl.dialogarena.soknadinnsending.consumer.kontaktinfo.dkif.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class DigitalKontaktinfoBolk {

    private DigitalKontaktinfo kontaktinfo;
    private Feil feil;

    @JsonCreator
    public DigitalKontaktinfoBolk(
            @JsonProperty("kontaktinfo") DigitalKontaktinfo kontaktinfo,
            @JsonProperty("feil") Feil feil) {
        this.kontaktinfo = kontaktinfo;
        this.feil = feil;
    }

    public DigitalKontaktinfo getKontaktinfo() {
        return kontaktinfo;
    }

    public Feil getFeil() {
        return feil;
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/personalia/telefonnummer/dto/DigitalKontaktinfoBolk.kt
package no.nav.sosialhjelp.soknad.personalia.telefonnummer.dto

data class DigitalKontaktinfoBolk(
    val kontaktinfo: Map<String, DigitalKontaktinfo>?,
    val feil: Map<String, Feil>?
)

data class DigitalKontaktinfo(
    val mobiltelefonnummer: String?
)

data class Feil(
    val melding: String?
)
